modules/core/src/main/java/org/locationtech/jts/geom/impl/CoordinateArraySequence.java

```diff
@@ -168,35 +171,7 @@ public CoordinateArraySequence(CoordinateSequence coordSeq)
    */
   protected Coordinate[] enforceArrayConsistency(Coordinate[] array)
   {
-     Coordinate sample = createCoordinate();
-     Class<?> type = sample.getClass();
-     boolean isConsistent=true;
-     for( int i = 0; i < array.length; i++) {
-       Coordinate coordinate = array[i];
-       if( coordinate != null && !coordinate.getClass().equals(type)) {
-         isConsistent = false;
-         break;
-       }
-     }
-     if( isConsistent ){
-       return array;
-     }
-     else {
-       Class<? extends Coordinate> coordinateType = sample.getClass();
-       Coordinate copy[] = (Coordinate[]) Array.newInstance(coordinateType, array.length);
-       for ( int i = 0; i < copy.length; i++){
-          Coordinate coordinate = array[i];
-          if( coordinate != null && !coordinate.getClass().equals(type)){
-            Coordinate duplicate = createCoordinate();
-            duplicate.setCoordinate(coordinate);
-            copy[i] = duplicate;
-          }
-          else {
-            copy[i] = coordinate;
-          }
-       }
-       return copy;
-     }
+     return Coordinates.enforceConsistency(array, dimension, measures);
```

Comment:
should this method be removed?  (from p2p conversation)

Reply:
Thanks you are correct it moved to CoordinateArrays utility class for client code to call if needed